libyul/optimiser/SimplificationRules.cpp

```diff
@@ -186,14 +186,14 @@ solidity::Instruction Pattern::instruction() const
 	return m_instruction;
 }
 
-Expression Pattern::toExpression(SourceLocation const& _location) const
+Expression Pattern::toExpression(SourceLocation const& _location, bool readable) const
 {
 	if (matchGroup())
 		return ASTCopier().translate(matchGroupValue());
 	if (m_kind == PatternKind::Constant)
 	{
 		assertThrow(m_data, OptimizerException, "No match group and no constant value given.");
-		return Literal{_location, assembly::LiteralKind::Number, YulString{formatNumber(*m_data)}, {}};
+		return Literal{_location, assembly::LiteralKind::Number, YulString{(readable == false ? formatNumber(*m_data) : formatNumberReadable(*m_data))}, {}};
```

Comment:
This cannot be used here, because Yul doesn't have arithmetic operators.

What we could do is leave an annotation which could be printed.

Reply:
I've omitted the format due to incompatibility.